File: src/main/java/com/akagiyui/drive/model/AvatarContent.java
package com.akagiyui.drive.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

/**
 * 头像内容
 *
 * @author AkagiYui
 */
@Setter
@Getter
@AllArgsConstructor
public class AvatarContent {
    private byte[] content;
    private String contentType;
}

File: src/main/kotlin/com/akagiyui/drive/model/AvatarContent.kt
package com.akagiyui.drive.model

/**
 * 头像内容
 *
 * @author AkagiYui
 */
data class AvatarContent(
    val content: ByteArray,
    val contentType: String,
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AvatarContent

        if (contentType != other.contentType) return false
        if (!content.contentEquals(other.content)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = content.contentHashCode()
        result = 31 * result + contentType.hashCode()
        return result
    }

}
